src/zenml/integrations/seldon/steps/seldon_deployer.py

```diff
@@ -386,3 +395,148 @@ def seldon_custom_model_deployer_step(
+    if not isinstance(experiment_tracker, MLFlowExperimentTracker):
+        raise get_missing_mlflow_experiment_tracker_error()
+
+    # fetch the MLflow model registry
+    model_registry = Client().active_stack.model_registry
+    if not isinstance(model_registry, MLFlowModelRegistry):
+        raise ValueError(
+            "The MLflow model registry step can only be used with an "
+            "MLflow model registry."
+        )
+
+    # fetch the model version
+    if params.registry_model_version:
+        model_version = model_registry.get_model_version(
+            name=params.registry_model_name,
+            version=params.registry_model_version,
+        )
+    elif params.registry_model_stage:
+        model_version = model_registry.get_latest_model_versions(
+            name=params.registry_model_name,
+            version_stages=[params.registry_model_stage],
+        )[0]
+
+    if not model_version:
+        raise ValueError(
+            f"No Model Version found for model name "
+            f"{params.registry_model_name} and version "
+            f"{params.registry_model_version} or stage "
+            f"{params.registry_model_stage}"
+        )
+    # Set the pipeline information from the model version
+    pipeline_name = getattr(
+        model_version.zenml_metadata, "zenml_pipeline_name", ""
+    )
+    pipeline_run_id = getattr(
+        model_version.zenml_metadata, "zenml_pipeline_run_id", ""
+    )
+    step_name = getattr(model_version.zenml_metadata, "zenml_step_name", "")
+
+    # update the step configuration with the real pipeline runtime information
+    params.service_config.pipeline_name = pipeline_name
+    params.service_config.pipeline_run_id = pipeline_run_id
+    params.service_config.pipeline_step_name = step_name
+
+    # fetch existing services with same pipeline name, step name and
+    # model name
+    existing_services = model_deployer.find_model_server(
+        pipeline_name=pipeline_name,
+        pipeline_step_name=step_name,
+        model_name=params.service_config.model_name,
+    )
+
+    # even when the deploy decision is negative, if an existing model server
+    # is not running for this pipeline/step, we still have to serve the
+    # current model, to ensure that a model server is available at all times
+    if existing_services:
+        service = cast(SeldonDeploymentService, existing_services[0])
+        # We need to start
+        # the previous model server if it is no longer running, to ensure that
+        # a model server is available at all times
+        if not service.is_running:
+            service.start(timeout=params.timeout)
+        return service
+
+    # invoke the Seldon Core model deployer to create a new service
+    # or update an existing one that was previously deployed for the same
+    # model
+    service_config = params.service_config
+    artifact_store_path = f"{Client().active_stack.artifact_store.path}/mlflow"
```

Comment:
Is this a path that the model registry uses to store the model? If yes, can we somehow have that code in the model registry implementation instead of in a seldon deployment step which knows about some internal mlflow registry details?